File: Web/ClientApp/src/api/ledger/DTOs/TransactionDto.ts
import {MoneyDto} from "../../value-history/DTOs/Money";

export type TransactionDto = {
    key: string;
    date: string;
    debit: LedgerEntryDto | undefined;
    credit: LedgerEntryDto | undefined;
}

export interface LedgerEntryDto {
    componentId: string;
    physicalAllocationId: string;
    value: MoneyDto
}

File: Core/DTOs/TransactionDto.cs
using FinanceTracker.Core.Primitives;

namespace FinanceTracker.Core.DTOs;

public record TransactionDto(
    Guid Key,
    DateOnly Date,
    LedgerEntryDto? Debit,
    LedgerEntryDto? Credit
);

public record LedgerEntryDto(
    Guid ComponentId,
    Money Value,
    Guid? PhysicalAllocationId
);